src/server.ts

```diff
@@ -103,6 +104,9 @@ io.on("connection", (socket: socketio.Socket) => {
 
     socket.on("act", (message: PlayerEvent, fn: (retval: any) => void) => {
         console.log(message);
+        if (Util.isRandomCiv(message.civilisation)) {
+            Util.setRandomCivilisation(message, draftId, draftsStore);
+        }
```

Comment:
I would replace this block with a single method call that
- checks if the civilisation is `RANDOM`
- replaces it with a valid option if that is the case
- returns the new `PlayerEvent`

Reply:
could look like
```javascript
message = Util.setRandomCivilisationIfNeeded(message, draftid, draftsStore);
```